src/nnet3/nnet-utils.h

```diff
@@ -53,6 +53,19 @@ int32 NumOutputNodes(const Nnet &nnet);
 /// returns the number of input nodes of this nnet.
 int32 NumInputNodes(const Nnet &nnet);
 
+/// This function works out the interval at which a particular named
+/// input (e.g. input_name == "input" or input_name == "ivector") is
+/// used.  If the input is not used at all this function returns -1;
+/// if it is potentially used on every frame (every "t" value) it returns 1
+/// if it is only used on frame 0 it returns 0; and if it is used only
+/// on multiples of n frames with n > 1, it returns n.
+int32 GetInputInterval(const Nnet &nnet, std::string input_name);
+
+/// internal function called in GetInputInterval().
+int32 GetInputIntervalInternal(const GeneralDescriptor &gen_desc,               
```

Comment:
In nnet-descriptor-test.cc, I call GetInputIntervalInternal(), rather than GetInputInterval() for testing, (because GetInputInterval() requires nnet as an input argument) . So I declare GetInputIntervalInternal() here for nnet-desciptor-test.cc to include.